stacker/providers/aws/default.py

```diff
@@ -106,18 +111,54 @@ def cloudformation(self):
 
         return self._cloudformation
 
-    def get_stack(self, stack_name, **kwargs):
-        try:
-            return retry_on_throttling(
-                self.cloudformation.describe_stacks,
-                kwargs=dict(StackName=stack_name))['Stacks'][0]
-        except botocore.exceptions.ClientError as e:
-            if "does not exist" not in e.message:
-                raise
+    @property
+    def listener(self):
+        # deals w/ multiprocessing issues w/ sharing ssl conns
+        # see https://github.com/remind101/stacker/issues/196
+        pid = os.getpid()
+        if pid != self._pid or not self._listener:
+            session = get_session(self.region)
+            self._listener = CloudListener(
+                LISTENER_NAME, session=session)
+            self._listener.start()
+
+        return self._listener
+
+    def flush_events(self):
+        logger.info("Purging old queue")
+        while 1:
+            messages = self.listener.get_messages()
+            if len(messages) == 0:
+                break
+
+            for message in messages:
+                message.delete()
```

Comment:
Should look at batch delete for this, rather than deleting each message one call at a time.